src/pages/Login.tsx

```diff
@@ -16,32 +24,69 @@ interface ILoginProps {
 interface ILoginState {
     email: string;
     password: string;
+    error?: string;
 }
 
 class LoginPage extends React.Component<ILoginProps, ILoginState> {
-    public state = {
-        email: "",
-        password: ""
-    };
+
+    componentWillMount() {
+      this.setState({
+        email: '',
+        password: ''
+      });
+    }
 
     private handleEmailAddressChange = (event: any) => {
-        this.setState({ email: event.target.value })
+        this.setState({ email: event.target.value });
     }
 
     private handlePasswordChange = (event: any) => {
-        this.setState({ password: event.target.value })
+        this.setState({ password: event.target.value });
     }
 
     private handleLogin = () => {
-        this.props.login(this.state);
+        this.setState({
+          ...this.state,
+          error: null // remove error message
+        }, () => {
+          Promise.all([
+              api.Auth.login(this.state.email, this.state.password),
+          ]).then(([ token ]) => {
+              // Authentification success
+              // login into app : setting user and token
+              this.props.login({
+                ...this.state,
+                token
+              });
+          }).then(() => {
+            // store user info
+            api.Members.byId(1)
+            .then((response) => {
+                this.props.setInfo(response);
+            });
+          }).catch((err) => {
+            // Error handler
+            console.log("err ", err);
+            this.setState({
+                ...this.state,
+                error: "Une erreur est survenue, veuillez réessayer."
+            });
+          });
+        });
+    }
+
+    private enterPress = (e: any) => {
```

Comment:
not a fan of the name. Not that I have a lot better to offer, but even "conditionalSubmit" or "submitOnEnter" feels more explicit